src/common/BrowserStorage.ts

```diff
@@ -158,36 +159,73 @@ export type CorrectItem = {
 	url: string;
 };
 
-export type Options = {
-	[K in keyof StorageValuesOptions]: Option<K>;
+export type OptionsDetails = {
+	[K in keyof StorageValuesOptions]: OptionDetails<StorageValuesOptions, K>;
 };
 
-export type Option<K extends keyof StorageValuesOptions> =
-	| SwitchOption<K>
-	| SelectOption<K>
-	| ListOption<K>;
-
-export type BaseOption<K extends keyof StorageValuesOptions> = {
+export type OptionDetails<T, K extends keyof T = keyof T> =
+	| SelectOptionDetails<T, K>
+	| SwitchOptionDetails<T, K>
+	| TextFieldOptionDetails<T, K>
+	| NumericTextFieldOptionDetails<T, K>
+	| CustomOptionDetails<T, K>;
+
+export type OptionDetailsByType<
+	T,
+	U extends OptionDetails<T, K>['type'],
+	K extends keyof T = keyof T
+> = U extends 'select'
```

Comment:
Yikes! I guess you used some time in getting this right?

Reply:
A little. Do you know of a better way to deal with this?

Reply:
Nope. It's good, just glad I didn't have to figure it out!